src/event-stream/event-server.ts

```diff
@@ -844,5 +848,11 @@ export async function startEventServer(opts: {
   const addrStr = typeof addr === 'string' ? addr : `${addr.address}:${addr.port}`;
   logger.info(`Event observer listening at: http://${addrStr}`);
 
-  return server;
+  const closeFn = async () => {
+    await new Promise<void>((resolve, reject) => {
+      logger.info('Closing event observer server...');
+      server.close(error => (error ? reject(error) : resolve()));
+    });
+  };
+  return Object.assign(server, { closeAsync: closeFn });
```

Comment:
```suggestion
  return { ...server, ...{ closeAsync: closeFn } };
```
Avoids possible unwanted mutation of `server`.

Reply:
Normally would 100% agree, however, in this case copying the enumerable object properties from a very stateful and complex class instance like node's `http.Server` into a new object, would likely result in a very broken result object. An intentional mutation here is ok for this specific case. The best option is probably just returning `{ server: server, closeAsync: closeFn }`.